src/ecs/wizards/executeCommand.ts

```diff
@@ -18,8 +18,25 @@ export interface CommandWizardState {
     confirmation?: 'yes' | 'suppress'
 }
 
-function assertValidRunningTask(t: any): t is ECS.Task & { taskArn: string } {
-    return t.taskArn !== undefined && t.lastStatus === 'RUNNING' && t.desiredStatus === 'RUNNING'
+function assertValidRunningTask(containerName: string) {
+    return function (t: ECS.Task): t is ECS.Task & { taskArn: string } {
+        let managed = false
+        if (t.containers && t.containers.length > 0) {
+            for (const c of t.containers) {
+                if (c.name === containerName) {
+                    if (c.managedAgents && c.managedAgents.length > 0) {
+                        for (const ma of c.managedAgents) {
+                            if (ma.name === 'ExecuteCommandAgent') {
+                                managed = true
```

Comment:
This assignment is pretty deeply nested. Could be written declaratively:
```ts
const managed = !!t.containers?.find(c => 
   c?.name === containerName && 
   c.managedAgents?.find(a => a.name === 'ExecuteCommandAgent')
)
```
Alternatively I'd use this pattern:
```ts
for (const c of (t?.containers ?? [])) {
...
}
```
If you prefer using for loops. While the declarative syntax isn't as performant, that hardly matters for such high-level code.